### PR TITLE
Fix printing & parsing of nested uncurried syntax.

### PR DESCRIPTION
**Pull request by IwanKaramazow**
Fixes https://github.com/facebook/reason/issues/1823

Parses/prints `add(. 1, .2)` correct as `(add(. 1))(. 2)`

### CODE REVIEW

**Comment by IwanKaramazow** [comment]
Still one case failing, looking into it now.

File: src/reason-parser/reason_parser.mly [hunk 1]
```diff
@@ -608,30 +608,96 @@ let process_underscore_application args =
...
+     *   we get the two nested Pexp_apply associated with
+     *   (add(. 2, 3))(. 4, 5)
+     *)
+    let rec make_appl body = function
+      | args::xs ->
+          let (uncurried, args) = args in
+          let expr = if args = [] then body
+          else
+            let (args, wrap) = process_underscore_application args in
+            let expr = mkexp (Pexp_apply (body, args)) in
```
Comment (cristianoc) [inline src/reason-parser/reason_parser.mly]: How about taking the location range from the arguments:
```
            let args_loc = match args, List.rev args with
              | ((_, s)::_), ((_, e)::_) -> mklocation s.pexp_loc.loc_start e.pexp_loc.loc_end
              | _ -> assert false in
            let expr = mkexp (Pexp_apply (body, args)) ~loc:args_loc in

```

Comment (IwanKaramazow) [reply src/reason-parser/reason_parser.mly]: Thanks for the code, exactly what was needed.

**Comment by chenglou** [comment]
M-M-MERGED

**Pull request merged**